_blog/scaling-dot-net-core-microservices/index.md

```diff
@@ -0,0 +1,403 @@
+The core focus has been to make the code readable to ubrew nderstand what is being achieved.
+To keep it simple, factors such as security have been left out, this will be addressed within future posts for best practices.
+
+## Case Study
+
+For our demo application, we're going to build a ticketing system.
+This system is going to be similar to a deli counter at a local market.
+When a customer comes to the website, they can request a ticket.
+When a ticket is generated, it is put into a message queue.
+There is a worker that can collect a ticket from the queue and then it takes 5 seconds to process that ticket.
+Once the ticket is processed, the customer is redirected to a success page.
+
+### Causes for concern
+
+When running the application locally, there are six different services that are required:
+
+1) Front end website
+2) API
+3) Database
+4) Message Queue
+5) Worker
+6) Worker monitor
+
+Looking at out application, lets assume that there is only a single instance of each of the services running.
+Is it a problem that the worker can only process a single ticket every 5 seconds?
+I'l be fine if we had a ticket once every 5 seconds, but usually people want to get a ticket at the same time.
+How long should we allow a user to wait prior to the ticket being completed?
+Should we scale the number of workers based on the number of tickets within the message queue?
+
+### Technology choices
+
+For this demo, all our code is written using C# and dotnet core.
+
+1) MVC Website
+2) MVC API
+3) Mongo DB
+4) RabbitMQ
+5) C# Console Application
+6) MVC Website
+
+### Running locally
+
+Using `tmux`, we can see 6 windows running each of the services we mentioned.
+All dotnet core services were started with `dotnet run` within the project directory.
+What happens if one of the services stops?  For example, let say that the worker service stops.
+The website would be able to submit a ticket to the API, and the API would put the ticket into the database.
+The API would then send the ticket to the queue, but nothing would be subscribed to pick up that ticket and process it.
+Also, we have a lot to do to get the service up and running locally, are we expecting to do this to debug the application locally?
+Is there a better way that we could manage this?
+
+## The Case For Containers
+
+So far, we've found out that running our microservice application has been a bit harder to do than we'd like.
+If we compare our services to spinning plates, we have a lot of spinning plates that are dependent on each other and we're not great at noticing when a plate has fallen.
+We're going to add a `Dockerfile` to each of our services to containerise them.
+
+### Containers should do just one thing, and one thing well
+
+Containers can be whatever you want them to be, a 30mb console application to a 4gb monolith.
+But they should really just do one job and do one job well.
```

Comment:
You say what containers are (bundles), but not why we should use them. The previous section was meant to address that, but... it's just telling me that it's hard to do microservices?
I think you could explain you could package different version of dotnet core or using other languages without worrying about what's inside.
> If you're a donet developer that is transitioning to open source technologies, it's great to have the option to mix and match languages.